Sources/Views/KMTagLabelsView.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  KMTagLabelsView.swift
+//  KommunicateChatUI-iOS-SDK
+//
+//  Created by sathyan elangovan on 21/11/22.
+//
+
+import Foundation
+
+class KMTagLabelsView: UIView {
+
+    var tagNames: [String] = [] {
+        didSet {
+            addTagLabels()
+        }
+    }
+    
+    let tagHeight:CGFloat = 30
+    let tagPadding: CGFloat = 16
+    let tagSpacingX: CGFloat = 8
+    let tagSpacingY: CGFloat = 8
+
+    var intrinsicHeight: CGFloat = 0
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        commonInit()
+    }
+    
+    func commonInit() -> Void {
+    }
+
+    func addTagLabels() -> Void {
+        // if we already have tag labels
+        //  remove any excess (e.g. we had 5 tags, new set is only 3)
+        while self.subviews.count > tagNames.count {
+            self.subviews.forEach({ $0.removeFromSuperview() })
```

Comment:
Why remove just the extras, not all? We are adding all of them again, no? Or should we reuse them? This can slow down the scroll if there are more number of threads. Removing and adding subviews in each cell is costly.

Reply:
I used this approach because max tag count will be 5 per conversation..also in tag view approach we had to remove&add subviews..